### PR TITLE
K8SPXC-1009 - Also enable super_read_only when replica

### PR DESCRIPTION
**Pull request by benlangfeld**
[K8SPXC-1009](https://jira.percona.com/browse/K8SPXC-1009) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

Without this change, users with SUPER can still perform DMLs against what should be read-only replicas.

It's not necessary to modify DisableReadonly() or IsReadonly() because enabling super_read_only implicitly enables read_only and disabling read_only implicitly disables super_read_only.

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy | passed |
| big-data | passed |
| upgrade-haproxy | passed |
| scheduled-backup | passed |
| scaling | passed |
| recreate | passed |
| storage | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | passed |
| limits | passed |
| demand-backup | passed |
| smart-update | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| affinity | passed |
| one-pod | passed |
| pitr | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| upgrade-consistency | passed |
| users | passed |
| demand-backup-encrypted-with-tls | failed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1094/commits/eada370a312bfbe8fe5c1edd8bea5c0346be5b48
image: `perconalab/percona-xtradb-cluster-operator:PR-1094-eada370a`

File: pkg/pxc/queries/queries.go [hunk 1]
```diff
@@ -230,8 +230,8 @@ func (p *Database) StopReplication(name string) error {
 }
 
 func (p *Database) EnableReadonly() error {
-	_, err := p.db.Exec("SET GLOBAL READ_ONLY=1")
-	return errors.Wrap(err, "set global read_only param to 1")
+	_, err := p.db.Exec("SET GLOBAL SUPET_READ_ONLY=1")
```
Comment (egegunes) [inline pkg/pxc/queries/queries.go:233]:
```suggestion
	_, err := p.db.Exec("SET GLOBAL SUPER_READ_ONLY=1")
```

**Review by egegunes — CHANGES_REQUESTED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| upgrade-haproxy | passed |
| haproxy | passed |
| big-data | passed |
| scaling | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | passed |
| limits | passed |
| monitoring-2-0 | passed |
| smart-update | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| upgrade-consistency | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1094/commits/30a6b3ea99b92623174d948de8af471516ebec4d
image: `perconalab/percona-xtradb-cluster-operator:PR-1094-30a6b3ea`

**Review by egegunes — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/recreate.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/haproxy.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/init-deploy.log) |
| restore-to-encrypted-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/restore-to-encrypted-cluster.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/demand-backup.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/limits.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/monitoring-2-0.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/affinity.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/pitr.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/one-pod.log) |
| auto-tuning | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/auto-tuning.log) |
| proxysql-sidecar-res-limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/proxysql-sidecar-res-limits.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/users.log) |
| demand-backup-encrypted-with-tls | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/demand-backup-encrypted-with-tls.log) |
| We run 14 out of 36|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1094/commits/74bfb5d906aa1712915dd29632f303eddbcc2a96
image: `perconalab/percona-xtradb-cluster-operator:PR-1094-74bfb5d9`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/storage.log) |
| upgrade-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/upgrade-haproxy.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/recreate.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/scaling.log) |
| scheduled-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/scheduled-backup.log) |
| cross-site | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/cross-site.log) |
| restore-to-encrypted-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/restore-to-encrypted-cluster.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/haproxy.log) |
| big-data | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/big-data.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/init-deploy.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/pitr.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/limits.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/monitoring-2-0.log) |
| demand-backup-encrypted-with-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/demand-backup-encrypted-with-tls.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/affinity.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/one-pod.log) |
| auto-tuning | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/auto-tuning.log) |
| proxysql-sidecar-res-limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/proxysql-sidecar-res-limits.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/users.log) |
| scaling-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/scaling-proxysql.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/self-healing.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/self-healing-chaos.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/security-context.log) |
| tls-issue-self | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/tls-issue-self.log) |
| self-healing-advanced | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/self-healing-advanced.log) |
| self-healing-advanced-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/self-healing-advanced-chaos.log) |
| upgrade-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/upgrade-proxysql.log) |
| tls-issue-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/tls-issue-cert-manager.log) |
| tls-issue-cert-manager-ref | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/tls-issue-cert-manager-ref.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/operator-self-healing.log) |
| operator-self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/operator-self-healing-chaos.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/smart-update.log) |
| validation-hook | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/validation-hook.log) |
| proxy-protocol | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/proxy-protocol.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1094/74bfb5d9/upgrade-consistency.log) |
| We run 36 out of 36|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1094/commits/74bfb5d906aa1712915dd29632f303eddbcc2a96
image: `perconalab/percona-xtradb-cluster-operator:PR-1094-74bfb5d9`

**Comment by benlangfeld** [comment]
@egegunes is it possible to get this merged please? This is very important for deployments which use replication.

**Comment by hors** [comment]
> @egegunes is it possible to get this merged please? This is very important for deployments which use replication.

@benlangfeld do not worry :) we are going to include your changes in next PXCO release

**Review by hors — CHANGES_REQUESTED** [review]
As I can see after these changes it is not possible to change password for `replication` user but it is common use case.

**Comment by benlangfeld** [comment]
> As I can see after these changes it is not possible to change password for `replication` user but it is common use case.

One could switch this off temporarily to make that change, leaving the node in a safe state the rest of the time.

**Comment by hors** [comment]
> > As I can see after these changes it is not possible to change password for `replication` user but it is common use case.
> 
> One could switch this off temporarily to make that change, leaving the node in a safe state the rest of the time.

yes, I think  we should

**Comment by benlangfeld** [comment]
> 
> 
> > > As I can see after these changes it is not possible to change password for `replication` user but it is common use case.
> 
> > 
> 
> > One could switch this off temporarily to make that change, leaving the node in a safe state the rest of the time.
> 
> 
> 
> yes, I think  we should 

So do you need a mention of that in the documentation or something else?

**Comment by hors** [comment]
> > > One could switch this off temporarily to make that change, leaving the node in a safe state the rest of the time.
> > 
> > 
> > 
> > yes, I think  we should 
> 
> So do you need a mention of that in the documentation or something else?
@benlangfeld I prefer to have this logic on operator end. E.g. Operator can detect that 
it's replica cluster and super_read_only is 1 and in case of password change set it to 0. As soon as the password is updated operator will change it back to 1. So, using this logic we always have read_only=1 and only temporary super_read_only=0

**Comment by benlangfeld** [comment]
> > > > One could switch this off temporarily to make that change, leaving the node in a safe state the rest of the time.
> > > 
> > > 
> > > yes, I think  we should
> > 
> > 
> > So do you need a mention of that in the documentation or something else?
> 
> @benlangfeld I prefer to have this logic on operator end. E.g. Operator can detect that it's replica cluster and super_read_only is 1 and in case of password change set it to 0. As soon as the password is updated operator will change it back to 1. So, using this logic we always have read_only=1 and only temporary super_read_only=0

Ah ok, you meant for operator-driven password changes. I'll see if I can find time to implement that.

**Comment by egegunes** [comment]
@benlangfeld I'm closing this PR due to inactivity. You can re-open it if you want to work on it. Please let us know if you need any help.

**Pull request closed without merging**